cmd/lotus-provider/deps/deps.go

```diff
@@ -236,18 +237,28 @@ Get it with: jq .PrivateKey ~/.lotus-miner/keystore/MF2XI2BNNJ3XILLQOJUXMYLUMU`,
 		deps.LW = sealer.NewLocalWorker(sealer.WorkerConfig{}, deps.Stor, deps.LocalStore, deps.Si, nil, wstates)
 	}
 	if len(deps.Maddrs) == 0 {
-		for _, s := range deps.Cfg.Addresses.MinerAddresses {
-			addr, err := address.NewFromString(s)
-			if err != nil {
-				return err
+		for _, s := range deps.Cfg.Addresses {
+			for _, s := range s.MinerAddresses {
+				addr, err := address.NewFromString(s)
+				if err != nil {
+					return err
+				}
+				deps.Maddrs = append(deps.Maddrs, dtypes.MinerAddress(addr))
```

Comment:
Should we check for duplicates and disallow them?

Reply:
done